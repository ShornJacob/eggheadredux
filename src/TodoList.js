import React from 'react'
import Todo from './Todo.js'

export default ({todos,onTodoClick}) => {
    
    // console.log(todos)
    // console.log(todos.length)
    // console.log(onTodoClick)


    
    return(
    <ul>
        {/* Todo returns a <li></li> */}
        {todos.map ( todo => <Todo 
                            key={todo.id}
                            {...todo}
                            onClick={ ()=> onTodoClick(todo.id)} /> )}
        
      </ul>
    )
}
