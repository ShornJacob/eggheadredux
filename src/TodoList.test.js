import React from 'react'
import TodoList from './TodoList'
import renderer from 'react-test-renderer';

it('renders correctly', () => {

   const todos = [{id:1}]
   const onTodoClick = () => null


    const tree = renderer
      .create(<TodoList todos={todos} onTodoClick={onTodoClick} /> )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });