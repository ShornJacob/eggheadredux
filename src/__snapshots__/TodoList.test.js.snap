// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<ul>
  <span>
    3
  </span>
</ul>
`;
